packages/design-picker/src/components/unified-design-picker.tsx

```diff
@@ -0,0 +1,469 @@
+					<DesignButtonContainer
+						key={ design.slug }
+						design={ design }
+						locale={ locale }
+						onSelect={ onSelect }
+						onPreview={ onPreview }
+						onUpgrade={ onUpgrade }
+						premiumBadge={ premiumBadge }
+						highRes={ false }
+						hideFullScreenPreview={ false }
+						hideDesignTitle={ false }
+						isPremiumThemeAvailable={ isPremiumThemeAvailable }
+						previewOnly={ previewOnly }
+						hasDesignOptionHeader={ hasDesignOptionHeader }
+						onCheckout={ onCheckout }
+					/>
+				) ) }
+			</div>
+		</div>
+	);
+};
+
+const GeneratedDesignPicker: React.FC< GeneratedDesignPickerProps > = ( {
+	locale,
+	designs,
+	verticalId,
+	onPreview,
+} ) => {
+	const isMobile = useViewportMatch( 'small', '<' );
+
+	return (
+		<div className="design-picker__grid">
+			{ designs.map( ( design, index ) => {
+				const previewUrl = getDesignPreviewUrl( design, {
+					language: locale,
+					vertical_id: verticalId,
+					viewport_width: isMobile ? MOBILE_VIEWPORT_WIDTH : DEFAULT_VIEWPORT_WIDTH,
+					viewport_height: DEFAULT_VIEWPORT_HEIGHT,
+					use_screenshot_overrides: true,
+				} );
+				return (
+					<div className="design-button-container" key={ `generated-design__${ index }` }>
+						<div className="design-picker__design-option">
+							<button className="generated-design-thumbnail" onClick={ () => onPreview( design ) }>
+								<span className="generated-design-thumbnail__image design-picker__image-frame design-picker__image-frame-no-header">
+									<ThemePreview
+										url={ previewUrl }
+										viewportWidth={ isMobile ? MOBILE_VIEWPORT_WIDTH : DEFAULT_VIEWPORT_WIDTH }
+									/>
+								</span>
+							</button>
+						</div>
+					</div>
+				);
+			} ) }
+		</div>
+	);
+};
+
+const UnifiedDesignPicker: React.FC< UnifiedDesignPickerProps > = ( {
+	locale,
+	onSelect,
+	onPreview,
+	onUpgrade,
+	verticalId,
+	staticDesigns,
+	generatedDesigns,
+	premiumBadge,
+	heading,
+	categorization,
+	previewOnly = false,
+	hasDesignOptionHeader = true,
+	isPremiumThemeAvailable,
+	onCheckout = undefined,
+} ) => {
+	const hasCategories = !! categorization?.categories.length;
+	const translate = useTranslate();
+	return (
+		<div
+			className={ classnames(
+				'design-picker',
+				`design-picker--theme-light`,
+				'design-picker__unified',
+				{
+					'design-picker--has-categories': hasCategories,
+				}
+			) }
+		>
+			{ heading }
+			{ generatedDesigns.length > 0 && (
+				<>
+					<div>
+						<h3> { translate( 'Custom designs for your site' ) } </h3>
+						<p className="unified-design-picker__subtitle">
+							{ translate( 'Based on your input, these designs have been tailored for you.' ) }
+						</p>
+					</div>
+					<GeneratedDesignPicker
+						locale={ locale }
+						designs={ generatedDesigns }
+						onPreview={ onPreview }
+						verticalId={ verticalId }
+					/>
+				</>
+			) }
+			<div>
+				<h3> { translate( 'Selected themes for you' ) } </h3>
```

Comment:
When there is no generated design, should we skip the following heading entirely? The copy looks literally redundant to me 😄

![image](https://user-images.githubusercontent.com/1525580/180404735-f8e2b5a3-8343-41b2-b128-28bc7e97b44f.png)

Reply:
I think you're right  👍